runtime/src/accounts.rs

```diff
@@ -1043,7 +1046,16 @@ impl Accounts {
 
             let (execution_result, nonce) = &execution_results[i];
             let maybe_nonce = match (execution_result, nonce) {
-                (Err(TransactionError::InstructionError(_, _)), Some(nonce)) => Some(nonce),
+                (Ok(_), Some(nonce)) => {
```

Comment:
Can we either combine this with the other `Ok(_)` match arm or constrain the other one with `None` for the `nonce` field?